libs/x-content/src/main/java/org/elasticsearch/common/xcontent/XContentParseException.java

```diff
@@ -45,8 +52,33 @@ public int getColumnNumber() {
         return location.map(l -> l.columnNumber).orElse(-1);
     }
 
+    @Nullable
+    public XContentLocation getLocation() {
+        return location.orElse(null);
+    }
+
     @Override
     public String getMessage() {
         return location.map(l -> "[" + l.toString() + "] ").orElse("") + super.getMessage();
     }
+
+    /**
+     * Return the detail message, including the message from the nested exception
+     * if there is one.
+     */
+    public String getDetailedMessage() {
```

Comment:
This seems to only be used for tests. Maybe it should be a helper method in the test framework instead of part of the public api? I would be afraid of something accidentally using this in ES code.

Reply:
I'm not sure, we have this also in `ElasticsearchException` (which is where I originally saw this). How about if instead I use `ExceptionsHelper.detailedMessage` since all of the callers are in `server`?

Reply:
The javadocs on that method show it is deprecated and should not be used. Why is this necessary? Why are tests converted in this PR to using it?

Reply:
I didn't convert the tests to use it, a fair number were already using `getDetailedMessage` provided by `ElasticsearchException`, because the previous `ParsingException` used to be a subclass of `ElasticsearchException`.

As for the necessity, it seems to make it easier for tests to check that somewhere in the chain of causes a specific exception message exists, without having to know exactly which cause it will be.

Reply:
Ok, my confusion stemmed from the fact the first result when searching down the diff was ScriptProcessorFactoryTests.java, which this PR changes from testing against `getMessage()`, to using `getDetailedMessage()`. I see now that is the only case. Can it be switched back?

Changing the tests to use `ExceptionsHelper.detailedMessage` seems ok given they all already use it.

Reply:
Okay, I've removed this method (getDetailedMessage) and reverted the ScriptProcessorFactoryTests back to the getMessage()